### PR TITLE
Merged upstream orientation/master with doximity/pseudo-master.

### PR DESCRIPTION
**Pull request by AndrewMcBurney**
https://www.pivotaltracker.com/n/projects/2007435/stories/143846435

This pull request merges upstream `orientation/master` with our fork `doximity/pseudo-master`.

A few important things to note:

1. The ruby version is now `2.4.0` instead of `2.3.1`. Due to this change - there exist a fair number of deprecation warnings.
2. JavaScript libraries are no longer installed with the command `npm install -g bower; bower install --allow-root`, but with `npm install` instead - I will update the wiki for this once the pull request has been merged in, and the changes are on production.
3. There exists some flakey (nondeterministic) tests which existed before merging the upstream branch in `deleting_an_author_spec.rb`, which still exist. More specifically, these tests are: 
```
rspec ./spec/features/deleting_an_author_spec.rb:54 # deleting an author author with articles, edits and outdatedness reports re-assigns report_outdated reports to a new user
rspec ./spec/features/deleting_an_author_spec.rb:30 # deleting an author author with articles, edits and outdatedness reports re-assigns articles to a new user
rspec ./spec/features/deleting_an_author_spec.rb:42 # deleting an author author with articles, edits and outdatedness reports re-assigns edits to a new user
```

### CODE REVIEW

**Comment by AndrewMcBurney** [comment]
Do we have CI setup to run tests when we have pull requests on this repository?

File: config/environments/staging.rb [hunk 1]
```diff
@@ -65,6 +65,10 @@
   # config.mandrill_mailer.default_url_options =
   config.action_mailer.default_url_options = { host: 'orientation-staging.herokuapp.com' }
 
+  #NOTE: config.mandrill_mailer.default_url_options =
```
Comment (brupm) [inline config/environments/staging.rb]: Bizarre spacing but if it's coming from the upstream, probably best to leave it alone and reduce conflicts.

Comment (AndrewMcBurney) [inline config/environments/staging.rb]: Oops, this is my fault - I added some notes to help me with the merging process, and I forgot to remove some of them. I'll add another commit to the PR to remove them. My apologies.

**Comment by ancooke** [comment]
Ready for deploy

**Pull request merged**